### PR TITLE
feat(compiler-cli): private i18n API for the CLI

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/@angular/compiler-cli/src/ngtools_api.ts [hunk 1]
```diff
@@ -124,4 +132,19 @@ export class NgTools_InternalApi_NG_2 {
...
+
+  /**
+   * @internal
+   * @private
+   */
+  static extractI18n(options: NgTools_InternalApi_NG2_ExtractI18n_Options): Promise<void> {
+    const hostContext: CompilerHostContext =
+        new CustomLoaderModuleResolutionHostAdapter(options.readResource, options.host);
+
+    // Create the Code Generator.
```
Comment (vicb) [inline modules/@angular/compiler-cli/src/ngtools_api.ts]: TODO: update

File: modules/@angular/compiler-cli/src/codegen.ts [hunk 2]
```diff
@@ -11,15 +11,13 @@
  * Intended to be used in a build step.
  */
 import * as compiler from '@angular/compiler';
```
Comment (hansl) [inline modules/@angular/compiler-cli/src/codegen.ts]: that's a random file to change...

File: modules/@angular/compiler-cli/src/extractor.ts [hunk 3]
```diff
@@ -15,27 +15,70 @@ import 'reflect-metadata';
 
 import * as compiler from '@angular/compiler';
```
Comment (hansl) [inline modules/@angular/compiler-cli/src/extractor.ts]: What's happening here? Did you also change this interface?

**Review by hansl — CHANGES_REQUESTED** [review]
There are changes to files but no mention of those in the PR description. Is that intentional?

**Review by hansl — APPROVED** [review]

File: modules/@angular/compiler-cli/src/extractor.ts (hunk 3 above)
Comment (vicb) [inline modules/@angular/compiler-cli/src/extractor.ts]: Added a comment in the commit message

**Pull request merged**